source/common/http/header_map_impl.cc

```diff
@@ -214,34 +127,19 @@ void HeaderString::setCopy(absl::string_view view) {
 }
 
 void HeaderString::setInteger(uint64_t value) {
-  switch (type_) {
-  case Type::Reference: {
-    // Switch back to inline and fall through.
-    type_ = Type::Inline;
-    buffer_.dynamic_ = inline_buffer_;
+  const uint32_t max_buffer_length = 32;
+  char inner_buffer[max_buffer_length];
+  string_length_ = StringUtil::itoa(inner_buffer, max_buffer_length, value);
```

Comment:
You should be able to itoa directly into the buffer after reserving it and avoid the copy?

Reply:
I've attempted the following:
`
BUFFER_IN_VEC_GET.resize(MaxIntegerLength);
const uint32_t int_length = StringUtil::itoa(BUFFER_IN_VEC_GET.data(), MaxIntegerLength, value);
BUFFER_IN_VEC_GET.resize(int_length);
`
but unfortunately it has ~100ns perf penalty (the first resize also initializes the 32 cells, and the second one "destroys" the unused ones).
If you think of a better way, let me know and I'll implement it and test it out.